src/storage/mvcc/reader.rs

```diff
@@ -17,6 +17,12 @@ use super::{Error, Result};
 use super::lock::Lock;
 use super::write::{Write, WriteType};
 
+#[derive(Clone, Default, Debug)]
+pub struct ScanMetrics {
```

Comment:
@siddontang Seems only suitable for the Scan command, which is rarely used in kv commands.